Realm/Realm/Linq/QueryObjects.cs

```diff
@@ -0,0 +1,150 @@
+﻿using System;
+
+namespace Realms
+{
+    public class QueryModel
+    {
+        public WhereClause WhereClause { get; set; }
+
+        public OrderingClause OrderingClause { get; set; }
+    }
+
+    public abstract class OrderingClause
+    {
+        public bool IsAscending { get; set; }
+
+        public bool IsReplacing { get; set; }
+
+        public string Property { get; set; }
+
+        public abstract string Kind { get; }
+    }
+
+
+    // TODO kind is not neccesarry, just the isAsc and isRep and property is needed
+    public class OrderByNode : OrderingClause
+    {
+        public override string Kind => "orderbyclause";
+    }
+
+    public class ThenByNode : OrderingClause
+    {
+        public override string Kind => "thenbyclause";
+    }
+
+    // Might be redundant
+    public class OrderByDescendingNode : OrderingClause
+    {
+        public override string Kind => "orderbydescclause";
+    }
+
+    // Might be redundant
+    public class ThenByDescendingNode : OrderingClause
+    {
+        public override string Kind => "thenbydescclause";
+    }
+
+    public class WhereClause
+    {
+        public ExpressionNode ExpNode { get; set; }
+
+        public string Kind { get; } = "whereclause";
```

Comment:
Here you are using an auto property, while later you are using an expression-bodied property (like on line 70). It should be uniformed (probably with the latter) :)